pkg/clusteragent/admission/mutate/common/test_utils.go

```diff
@@ -357,7 +360,19 @@ func FakeConfig(t *testing.T) model.BuildableConfig {
 func FakeConfigWithValues(t *testing.T, values map[string]interface{}) model.BuildableConfig {
 	mockConfig := configmock.New(t)
 	for k, v := range values {
+		// Setting config directly on the mock requires basic types which are not easy to use with targets.
+		if k == "apm_config.instrumentation.targets" {
```

Comment:
This is admittedly the weirdest part of this PR. I want to be able to assign targets from a list of structs, but `SetWithoutSource` only takes primitive types.

Reply:
Do we reference this or other similar keys elsewhere? Would nice to be `const` if it's used more than once to avoid "magic" values.